packages/cli/src/commands/generate.ts

```diff
@@ -81,38 +90,134 @@ async function outputCodegen(
     path.join(outputFolder, "common-types.ts"),
     typesFileContent,
   )
-  await fs.writeFile(
-    path.join(outputFolder, "public-types.ts"),
-    `export { ${publicTypes.join(",\n")} } from './common-types';`,
-  )
   await Promise.all(
     chains.map((chain, i) =>
-      createDtsFile(chain.key, outputFolder, descriptorsFileContent[i]),
+      fs.writeFile(
+        join(outputFolder, `${chain.key}.ts`),
+        descriptorsFileContent[i],
+      ),
     ),
   )
+  await generateIndex(
+    outputFolder,
+    chains.map((chain) => chain.key),
+    publicTypes,
+  )
+}
+
+async function compileCodegen(packageDir: string) {
+  const srcDir = join(packageDir, "src")
+  const outDir = join(packageDir, "dist")
+
+  await tsup.build({
+    format: ["cjs", "esm"],
+    entry: [path.join(srcDir, "index.ts")],
+    outDir,
+    outExtension: (ctx) => ({
+      js: ctx.format === "esm" ? ".mjs" : ".js",
+    }),
+  })
 
-  tsc.build({
-    basePath: outputFolder,
+  await tsc.build({
+    basePath: srcDir,
     compilerOptions: {
       skipLibCheck: true,
-      emitDeclarationOnly: true,
       declaration: true,
+      emitDeclarationOnly: true,
       target: "esnext",
       module: "esnext",
       moduleResolution: "node",
       resolveJsonModule: true,
       allowSyntheticDefaultImports: true,
+      outDir,
     },
   })
-}
 
-const createDtsFile = async (key: string, dest: string, code: string) => {
-  const tscFileName = path.join(dest, key)
+  // const esmDir = join(packageDir, "esm")
+  // const cjsDir = join(packageDir, "cjs")
+
+  // if (existsSync(esmDir)) await fs.rm(esmDir, { recursive: true })
+  // if (existsSync(cjsDir)) await fs.rm(cjsDir, { recursive: true })
 
-  if (await fsExists(`${tscFileName}.ts`)) await fs.rm(`${tscFileName}.ts`)
+  // console.log("Generating ESM")
+  // tsc.build({
+  //   basePath: srcDir,
+  //   compilerOptions: {
+  //     skipLibCheck: true,
+  //     declaration: true,
+  //     target: "esnext",
+  //     module: "esnext",
+  //     moduleResolution: "node",
+  //     resolveJsonModule: true,
+  //     allowSyntheticDefaultImports: true,
+  //     outDir: join(packageDir, "esm"),
+  //   },
+  // })
+
+  // console.log("Generating CJS")
+  // tsc.build({
+  //   basePath: srcDir,
+  //   compilerOptions: {
+  //     skipLibCheck: true,
+  //     declaration: false,
+  //     target: "esnext",
+  //     module: "commonjs",
+  //     moduleResolution: "node",
+  //     resolveJsonModule: true,
+  //     allowSyntheticDefaultImports: true,
+  //     outDir: join(packageDir, "cjs"),
+  //   },
+  // })
+  // const nonSourceFiles = (await fs.readdir(srcDir)).filter(
+  //   (file) => !file.endsWith(".ts"),
+  // )
+  // await Promise.all(
+  //   nonSourceFiles.flatMap((file) => [
+  //     fs.copyFile(join(srcDir, file), join(esmDir, file)),
+  //     fs.copyFile(join(srcDir, file), join(cjsDir, file)),
+  //   ]),
+  // )
+}
```

Comment:
Do you want to keep this around just in case that `tsup` ends up being problematic?